keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/DoubleScalarTensorShapePreservationTest.java

```diff
@@ -0,0 +1,230 @@
+package io.improbable.keanu.tensor.dbl;
+
+import io.improbable.keanu.tensor.Tensor;
+import org.junit.Before;
+import org.junit.Test;
+import org.nd4j.linalg.api.shape.Shape;
+
+import static org.junit.Assert.assertArrayEquals;
+
+public class DoubleScalarTensorShapePreservationTest {
+
+    private static DoubleTensor doubleTensor;
+    private static DoubleTensor scalarDoubleTensor;
+
+    @Before
+    public void setup() {
+        doubleTensor = DoubleTensor.create(new double[]{1, 2, 3, 4, 5, 6}, new long[]{2, 3});
+        scalarDoubleTensor = DoubleTensor.create(1.0, new long[]{1, 1, 1, 1});
+    }
+
+    @Test
+    public void doubleScalarMultiplicationPreservesShape() {
+        Tensor multiplicationResult = doubleTensor.times(scalarDoubleTensor);
+        resultShapeMatchesBroadcastShape(multiplicationResult, doubleTensor, scalarDoubleTensor);
+    }
+
+    @Test
+    public void doubleScalarAdditionPreservesShape() {
+        Tensor additionResult = doubleTensor.plus(scalarDoubleTensor);
+        resultShapeMatchesBroadcastShape(additionResult, doubleTensor, scalarDoubleTensor);
+    }
+
+    @Test
+    public void doubleScalarDivisionPreservesShape() {
+        Tensor divisionResult = doubleTensor.div(scalarDoubleTensor);
+        resultShapeMatchesBroadcastShape(divisionResult, doubleTensor, scalarDoubleTensor);
+    }
+
+    @Test
+    public void doubleScalarSubtractionPreservesShape() {
+        Tensor subtractionResult = doubleTensor.minus(scalarDoubleTensor);
+        resultShapeMatchesBroadcastShape(subtractionResult, doubleTensor, scalarDoubleTensor);
+    }
+
+    @Test
+    public void doubleScalarMultiplicationPreservesShapeReverse() {
```

Comment:
nit: put the two Multiplication tests next to each other, to match the pattern in the rest of the file

likewise for Addition, Division, Subtraction

Reply:
done